src/doc/rustdoc/src/linking-to-items-by-name.md

```diff
@@ -89,6 +114,32 @@ struct Foo {}
 fn Foo() {}
 ```
 
-Note: Because of how `macro_rules!` macros are scoped in Rust, the intra-doc links of a `macro_rules!` macro will be resolved [relative to the crate root][#72243], as opposed to the module it is defined in.
+## Warnings, re-exports, and scoping
+
+Links are resolved in the current module scope, even when re-exported. If a link from another
+crate fails to resolve, no warning is given.
+
+When re-exporting an item, rustdoc allows additional documentation to it. That documentation will
+be resolved in the new scope, not the original, allowing you to link to items in the current
+crate. The new links will still give a warning if they fail to resolve.
```

Comment:
I would say something along the lines of (wording could be improved):

```suggestion
be resolved in the scope of the re-export, not the original, allowing you to link to items in the crate that's re-exporting.
The new links will still give a warning if they fail to resolve.
```

to be extra-clear what "new" and "current" mean.